src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,204 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.exeptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.DbConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders AS o"
+                        + " JOIN orders_products AS op ON o.order_id = op.orders_id"
+                        + " JOIN products AS p ON p.product_id = op.products_id"
+                        + " WHERE o.user_id = ? AND o.deleted = false ";
+
+        List<Order> orderList = new ArrayList<>();
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            while (sqlResult.next()) {
+                Long id = sqlResult.getLong("order_id");
+                Order order = new Order(id, userId);
+                orderList.add(order);
+            }
+            for (Order order : orderList) {
+                order.setProducts(extractProductForOrder(order.getId(), connection));
```

Comment:
You can split your requests into the following two: first - get user and product_id (only!) and close connection. Second - has method with connection to get all product details.

Reply:
I tried to use minim connections :)